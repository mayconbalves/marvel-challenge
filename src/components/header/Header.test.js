import { render, screen, fireEvent } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import Header from './'

describe('<Header />', () => {
  const mockStore = configureStore([thunk])
  let store
  store = mockStore({
    charactersReducer: { characters: [] }
  })

  it('should render correct name of StyledInput', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <Header />
        </Provider>
      </MemoryRouter>
    )

    expect(
      screen.getByRole('textbox', { Name: 'hero' })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
