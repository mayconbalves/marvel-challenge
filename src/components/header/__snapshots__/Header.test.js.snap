// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correct name of StyledInput 1`] = `
<div
  class="sc-gsTCUz FUAyX"
>
  <div
    class="sc-hKgILt eaPfCO"
  >
    <a
      href="/"
    >
      <img
        alt="logo"
        class="sc-dlfnbm gWQjsB"
        src="logo.png"
      />
    </a>
  </div>
  <div
    class="sc-eCssSg fTTyBU"
  >
    <input
      class="sc-bdfBwQ hFmnHx"
      placeholder="Procure seu herÃ³i"
      value=""
      width="80%"
    />
  </div>
</div>
`;
